src/DotLiquid/Util/ReflectionCache.cs

```diff
@@ -0,0 +1,45 @@
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+using System.Runtime.CompilerServices;
+
+namespace DotLiquid.Util
+{
+    internal class ReflectionCache
```

Comment:
nit: XML comments for the class, which is an opportunity to document for code maintainers why we have this cache class.